Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -804,48 +804,210 @@ Another compliant solution is to define the API level to 17 (JELLY_BEAN_MR1) and
+- `import com.j256.db`
+- `import com.j256.stmt`
+- `import com.j256.table`
+
+Please make sure that logging is disabled.
+
+**`SugarORM`**
+Search the source code for the following keywords:
+- `import com.github.satyan`
+- `extends SugarRecord<Type>`
+- in the AndroidManifest, there will be `meta-data` entries with values such as `DATABASE`, `VERSION`, `QUERY_LOG` and `DOMAIN_PACKAGE_NAME`. 
+
+Make sure that `QUERY_LOG` is set to false.
+
+**`GreenDAO`**
+Search the source code for the following keywords:
+
+- `import org.greenrobot.greendao.annotation.Convert`
+- `import org.greenrobot.greendao.annotation.Entity`
+- `import org.greenrobot.greendao.annotation.Generated`
+- `import org.greenrobot.greendao.annotation.Id`
+- `import org.greenrobot.greendao.annotation.Index`
+- `import org.greenrobot.greendao.annotation.NotNull`
+- `import org.greenrobot.greendao.annotation.*`
+- `import org.greenrobot.greendao.database.Database`
+- `import org.greenrobot.greendao.query.Query`
+
+
+**`ActiveAndroid`**
+Search the source code for the following keywords:
+
+- `ActiveAndroid.initialize(<contextReference>);`
+- `import com.activeandroid.Configuration`
+- `import com.activeandroid.query.*`
+
+**`Realm`**
+Search the source code for the following keywords:
+
+- `import io.realm.RealmObject;`
+- `import io.realm.annotations.PrimaryKey;`
+
+
+##### Parcelable
+Verify that, when sensitive information is stored in an Intent using a Bundle containing a Parcelable, then appropriate security measures are taken. Make sure to use explicit intents and reassure proper additional security controls in case of application level IPC (e.g. signature verification, intent-permissions, crypto).
 
 
 #### Dynamic Analysis
+There are various steps one can take to do dynamic analysis:
 
--- TODO [Create content for dynamic analysis of "Testing Object (De-)Serialization" ] --
+1. Regarding the actual persistence: use the techniques described in the data storage chapter.
+2. Regarding the reflection based approaches: use Xposed to hook into the de-serialization methods or add extra unprocessable information to the serialized objects to see how they are handled (e.g.: will the application crash? Or can you extract extra information by enriching the objects?)
 
 #### Remediation
+There are a few generic remediation steps one can always take:
 
--- TODO [Describe the best practices that developers should follow to prevent this issue "Testing Object (De-)Serialization".] --
+1. Make sure that sensitive data after serialization/persistance has been encrypted and HMACed/signed. See the crypto chapter for more details.
+2. Make sure that keys used for step 1 cannot be extracted easily. See the storage data chapter for more details.
+3. Make sure that you only use `Serializable` in case a class remains stable.
+4. Dont use reflection based persistence (e.g. do not use reflection based serializable, JSON based libraries or ORM libraries) in case of high risk applications.
```

Comment:
I don't think this chapter has made it clear why reflection based persistence would be a security issue.

> Dont use

_Don't_ use

Also, maybe you can separate the two sentences to make it more clear:

> Don't use reflection based persistence in case of high risk applications. For example, do not use reflection based serializable, JSON based libraries or ORM libraries.